cristin-commons/src/main/java/no/unit/nva/utils/AccessUtils.java

```diff
@@ -72,4 +84,20 @@ private static boolean clientIsInternalBackend(RequestInfo requestInfo) {
     private static boolean requesterHasNoAccessRightToEditProjects(RequestInfo requestInfo) {
         return !requestInfo.getAccessRights().contains(EDIT_OWN_INSTITUTION_PROJECTS);
     }
+
+    private static boolean requesterHasAccessRightInBearerToken(RequestInfo requestInfo) {
+        try {
+            final var authorizationHeader = requestInfo.getHeaders().get(AUTHORIZATION);
+            if (nonNull(authorizationHeader) && authorizationHeader.startsWith(BEARER)) {
+                var token = authorizationHeader.substring(BEARER.length() + 1, authorizationHeader.length());
+                var payload = new String(Base64.getUrlDecoder().decode(token.split(DOT_SEPARATOR)[PAYLOAD_SEGMENT]));
+                var accessRights = JsonUtils.dtoObjectMapper.readTree(new StringReader(payload))
+                        .get(CUSTOM_ACCESS_RIGHTS).asText();
+                return accessRights.contains(EDIT_OWN_INSTITUTION_USERS);
+            }
```

Comment:
Can you use https://github.com/auth0/java-jwt to *verify* and extract the relevant parts?

Reply:
Verifying token with keys from Cognito